style/js/shield_defs.js

```diff
@@ -54,6 +55,13 @@ function roundedRectShield(
   };
 }
 
+function banneredShield(baseDef, modifiers) {
+  let newShield = Util.cp(baseDef);
+  newShield.backgroundImage = baseDef.backgroundImage;
```

Comment:
Redundant?

Reply:
So `backgroundImage` in this case is a pointer to the sprites object in maplibre and doesn't survive a stringify copy - that's why I put that there.  Is there a better way perhaps to change that pointer to a string that can be a lookup into the sprite sheet maybe?  🤔

Reply:
Ah, I see there's a Uint8Array in there.

Sure, you could change the backgroundImage to be an array of string keys into shieldImages.

As is would be fine with a comment.

Reply:
I think you're probably also clobbering textLayoutConstraint.

Reply:
Ah yes that's true, and not apparently because no bannered route uses an unusual text layout constraint.  I'm wondering if the shield definition objects should be more portable or if we don't really care.

Reply:
Same suggestion for the constraints applies, either reference them by name to allow copying in this manner, or comment that they need to be reinstated.

Or switch to TypeScript and let the compiler do my job for me :)

Reply:
> Or switch to TypeScript and let the compiler do my job for me :)

Uhh... 😅